sw/device/silicon_creator/lib/drivers/pwrmgr.c

```diff
@@ -23,13 +24,27 @@ static_assert(
     PWRMGR_RESET_EN_EN_FIELD_WIDTH == 1,
     "RESET_EN field width changed, kAllResetsEnable may need to be updated");
 
+void pwrmgr_cdc_sync(uint32_t n) {
+  // We want to timeout if the CDC bit doesn't self clear.  We calculate the
+  // number of CPU cycles of 1 clock in the AON domain and add an additional
+  // 25 percent.
+  uint32_t cpu_cycle_timeout =
+      (uint32_t)kClockFreqCpuHz / (uint32_t)kClockFreqAonHz;
+  cpu_cycle_timeout += cpu_cycle_timeout / 4;
```

Comment:
That's not nearly long enough. It takes up to 3 AON cycles + 2 "fast domain" cycles after the store is delivered to pwrmgr for the sync to make it into the AON domain, then another 4 "fast domain" cycles + 1 AON cycle to go back in the other direction.

The "fast domain" cycles will likely be absorbed by the CPU and interconnect, but I'd probably go with at least the 4 AON cycles here (maybe 5, for additional margin for future tops).

Reply:
Done.  There are 4 uses of the CDC_SYNC operation in the ROM:

1. One in rom_start.S during some early init.  It just sets the register and doesn't wait.
2. Watchdog configuration synchronizes twice.
3. Pwrmgr synchronizes once after enabling all reset sources.
4. Sensor_ctrl synchronizes after configuring the sensor severity.  I've been instructed that I need to wait "20 clocks in the slow domain".  As such, sensor_ctrl synchronizes 5 times.